packages/ai-jsx/src/stream/index.ts

```diff
@@ -99,6 +99,27 @@ export function toStreamResponse(renderable: Renderable): Response {
   );
 }
 
+/**
+ * Converts a {@link Renderable} to a {@link ReadableStream} that will stream the rendered
+ * content as UTF-8 encoded text.
+ */
+export function toTextStream(renderable: Renderable): ReadableStream<Uint8Array> {
+  let previousValue = '';
+  const generator = createRenderContext().render(renderable, { appendOnly: true })[Symbol.asyncIterator]();
+  return new ReadableStream({
+    async pull(controller) {
+      const next = await generator.next();
+      const delta = next.value.slice(previousValue.length);
```

Comment:
Did we consider an alternative where `appendOnly` causes the generator to return deltas directly?

Reply:
I've previously tried to go down similar paths and wasn't sure I could make it work in time :)

I have a (shelved, for now) branch that rewrites rendering and will give code that wants it (e.g. this, nested UI/AI streams, partial rendering) much finer-grained visibility/control over rendering so that we can actually propagate fine-grained deltas out of the tree. (Currently the in-progress tree is only implicitly represented on co-routine callstacks which makes things tricky.) Depending on priorities I might take a stab at finishing that when I get back.